### PR TITLE
chore(internal/postprocessor): some refactoring and fixes

### PR DESCRIPTION
**Pull request by codyoss**
- Updates tests to read from testdata files
- Fixing some formatting issues that could arise in updated commit messages
- Fixes linked bug
- Updates lockfile to use this code

Fixes: #7474

### CODE REVIEW

**Comment by codyoss** [comment]
Tried to split this up into logical commits to make it easier to review. I will have a followup PR to clean up more style things

File: internal/postprocessor/main.go [hunk 1]
```diff
@@ -686,23 +684,22 @@ func extractHashFromLine(line string) string {
...
+	titleParts := strings.Split(title, ":")
+	commitPrefix := titleParts[0]
+	msg := strings.TrimSpace(titleParts[1])
 
-	titleSlice := strings.Split(title, ":")
-	firstTitlePart := titleSlice[0]
-	secondTitlePart := strings.TrimSpace(titleSlice[1])
-
-	if strings.HasSuffix(firstTitlePart, "!") {
+	// If a scope is already provided, remove it.
```
Comment (quartzmo) [inline internal/postprocessor/main.go:692]: Is there a test for this? I looked through the `.input` files and didn't see a scope, but I may have missed it.

Comment (codyoss) [reply internal/postprocessor/main.go:692]: The last test-case is net-new and catches this case. Because the scope was being duplicated in PRs 😄

Comment (quartzmo) [reply internal/postprocessor/main.go:692]: Looks good!

**Review by quartzmo — APPROVED** [review]

**Pull request merged**